kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMSettingsConverter.java

```diff
@@ -291,14 +341,17 @@ public static Map<String, Variant<?>> build80211WirelessSecuritySettings(Network
         KuraWifiSecurityType securityType = KuraWifiSecurityType.fromString(
                 props.get(String.class, KURA_PROPS_KEY_WIFI_SECURITY_TYPE, deviceId, propMode.toLowerCase()));
 
-        if (securityType == KuraWifiSecurityType.SECURITY_WEP) {
-            return createWEPSettings(props, deviceId, propMode);
-        } else if (securityType == KuraWifiSecurityType.SECURITY_WPA
-                || securityType == KuraWifiSecurityType.SECURITY_WPA2
-                || securityType == KuraWifiSecurityType.SECURITY_WPA_WPA2) {
-            return createWPAWPA2Settings(props, deviceId, propMode);
-        } else {
-            throw new IllegalArgumentException("Security type \"" + securityType + "\" is not supported.");
+        switch (securityType) {
```

Comment:
@mattdibi 

this has been bugging me for so long. I really really really needed to change it to a switch statement.

are there any caveats I should be aware of by doing this?

Reply:
> I really really really needed to change it to a switch statement.

I know and agree with your decision :)

> are there any caveats I should be aware of by doing this?

None that I can think of. LGTM